plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,245 @@
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.docker.internal;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.validation.ConstraintViolation;
+
+import org.sonatype.nexus.bootstrap.jetty.ConnectorConfiguration;
+import org.sonatype.nexus.bootstrap.jetty.ConnectorRegistrar;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.config.ConfigurationFacet;
+import org.sonatype.nexus.repository.docker.DockerConnectorFacet;
+import org.sonatype.nexus.repository.manager.RepositoryManager;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.validation.ConstraintViolations;
+import org.sonatype.nexus.validation.ManualValidator;
+import org.sonatype.nexus.validation.constraint.PortNumber;
+
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConfiguration.Customizer;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.ServerConnector;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @since 3.0
+ */
+@Named
+public class DockerConnectorFacetImpl
+    extends FacetSupport
+    implements DockerConnectorFacet
+{
+  static final String CONFIG_KEY = "docker";
+
+  private final RepositoryManager repositoryManager;
+
+  private final ManualValidator manualValidator;
+
+  private final ConnectorRegistrar connectorRegistrar;
+
+  private Config config;
+
+  private ServerConnector jettyConnector;
+
+  private ConnectorConfiguration httpConfig;
+
+  private ConnectorConfiguration httpsConfig;
+
+  @Inject
+  public DockerConnectorFacetImpl(final RepositoryManager repositoryManager,
+                                  final ManualValidator manualValidator,
+                                  final ConnectorRegistrar connectorRegistrar)
+  {
+    this.repositoryManager = checkNotNull(repositoryManager);
+    this.manualValidator = checkNotNull(manualValidator);
+    this.connectorRegistrar = checkNotNull(connectorRegistrar);
+  }
+
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    facet(ConfigurationFacet.class).validateSection(configuration, CONFIG_KEY, Config.class);
+    Config config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "config" which hides the field declared at line 61. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)